python/python/lance/torch/async_dataset.py

```diff
@@ -69,7 +69,10 @@ def __iter__(self):
 
     def close(self):
         self.shutdown.value = True
-        for _ in self:
+        try:
+            for _ in self:
+                pass
+        except Exception:
```

Comment:
`ruff` does not complaint? 😵‍💫

Reply:
Ha! This is not a bare expect. As long as there is some type the linter is happy. I think this is meant to prevent people from catching non error types